ui-asset-management/api/svc-asset-related-assets/extension/clients/related_assets_client.go

```
@@ -101,32 +90,64 @@ func (c *RelatedAssetsClient) GetRelatedAssetsDetails(ctx context.Context, tenan
-				documentId := relatedAssets[docId].(string)
-				assetType := relatedAssets[docType].(string)
-				resourceId := relatedAssets[_resourceId].(string)
-				for i, relatedAsset := range allRelatedAssets {
-					if relatedAsset.AssetType == assetType && relatedAsset.ResourceId == resourceId {
-						allRelatedAssets[i].AssetId = documentId
-						allRelatedAssets[i].AssetTypeName = assetTypeName
-					}
+	// fetch related parent assets
+	parentRelatedAssetDetails, err := c.elasticSearchClient.FetchParentRelatedAssets(ctx, tenantId, allSources, assetId)
+	if err != nil {
+		fmt.Errorf("failed to fetch parent related asset details %w", err)
```

golangci-lint (1.62.2)

Error return value of `fmt.Errorf` is not checked

(errcheck)